src/synthesizer/blackholes.py

```diff
@@ -0,0 +1,460 @@
+        
+        return spectra
+
+    def _get_spectra_lr(self, parameter_dict, line_region):
+        """
+        Generate the spectra of a generic line region.
+
+        Arguments
+            parameter_dict (dict)
+                A dictionary of all the parameters.
+            line_region (str)
+                The specific line region, i.e. 'nlr' or 'blr'.
+            
+        Returns:
+            synthesizer.sed.Sed
+                The NLR spectra
+        """
+
+        # In the Unified AGN model the NLR/BLR is illuminated by the isotropic 
+        # disc emisison hence the need to replace this parameter if it exists.
+        # Not all models require an inclination though.
+        if 'cosine_inclination' in parameter_dict.keys():
+            parameter_dict['cosine_inclination'] = 0.5
+
+        # get the grid point
+        grid_point = self._get_grid_point(parameter_dict, line_region)
+
+        # get the nebular spectra of the line region
+        sed = (parameter_dict[f'covering_fraction_{line_region}'] *
+               self.grid[line_region].get_spectra(grid_point,
+                                                  spectra_id='nebular'))
+
+        return sed
+
+    def _get_spectra_torus(self, parameter_dict):
+        """
+        Generate the torus emission Sed.
+
+        Arguments
+            parameter_dict (dict)
+                A dictionary of all the parameters.
+
+        Returns
+            spectra (dict, synthesizer.sed.Sed)
+                Dictionary of synthesizer Sed instances.
+        """
+
+        # In the Unified AGN model the torus is illuminated by the isotropic
+        # disc emisison hence the need to replace this parameter if it exists.
+        # Not all models require an inclination though.
+        if 'cosine_inclination' in parameter_dict.keys():
+            parameter_dict['cosine_inclination'] = 0.5
+     
+        # Get NLR grid points
+        nlr_grid_point = self._get_grid_point(parameter_dict, 'nlr')
+
+        # Calcualte the disc emission, since this is incident it doesn't matter
+        # if we use the NLR or BLR grid as long as we use the correct grid 
+        # point.
+        disc_spectra = self.grid['nlr'].get_spectra(nlr_grid_point,
+                                            spectra_id='incident')
+
+        # calculate the bolometric dust lunminosity as the difference between
+        # the intrinsic and attenuated
+        torus_bolometric_luminosity = (
+            (parameter_dict['theta_torus']/(90*deg)) *
+            disc_spectra.measure_bolometric_luminosity())
+
+        # get the spectrum and normalise it properly
+        lnu = torus_bolometric_luminosity.to("erg/s").value * \
+            self.torus_emission_model.lnu(disc_spectra.lam)
+
+        # create new Sed object containing torus spectra and return it
+        return Sed(disc_spectra.lam, lnu=lnu)
+
+    def get_spectra(self, spectra_ids=None, **kwargs):
```

Comment:
I still think this should also be implemented like the stellar `generate_lnu` methods on components but I see why you might want to be able to do it on the "`Grid`" analog.